### ISSUE TITLE
It blocks waiting for stdin

### ISSUE THREAD

**Issue by Rudxain** [RazrFalcon/cargo-bloat#90]
I ran the usage example and many variants of it, on both the crate root AND the `target/release` dirs. Why does this happen?

**Comment by RazrFalcon** [comment]
Can you provide more info?

**Comment by Rudxain** [comment]
I ran this line:
```sh
cargo bloat --release
```
With WD at root of my crate.
Also tried:
```sh
cargo bloat --release -n 10
```
I ran it before and after running `cargo build -r`

OS: Linux Mint Cinnamon 20.3
CPU: x86 64bit
Rust edition: 2021
Cargo and `rustup` are updated (I'll come back in a moment with specific version numbers, please wait).

More details:
I assume it waits for `stdin` because running any of those cmds blocks the program, but I can still type text in the terminal (without shell access, until I press Ctrl+c, then I gain shell access again). crtl+d doesn't help. And adding invalid args makes the program reply with "unused argument"

I followed the usual install route:
```sh
cargo install cargo-bloat
```

**Comment by RazrFalcon** [comment]
So `cargo build` works but `cargo bloat` doesn't? This is very strange. I have no idea why it could be happening.
Maybe you have some environment variables set? What about:

```
echo $RUSTC_WRAPPER
echo $RUSTFLAGS
```

Otherwise I could only assume that something is wrong on your side and I don't know what.

**Comment by Rudxain** [comment]
> Maybe you have some environment variables set? What about:
> ```
> echo $RUSTC_WRAPPER
> echo $RUSTFLAGS
> ```

Both are unset or empty. Don't worry, it works now! I had to wait for 1 or 2 whole minutes before the first "compiling" line appeared. The compilation was fast, but the latency *before* the compilation was big. Maybe it's a profile config I set, I'll try disabling everything to default

**Comment by RazrFalcon** [comment]
The first thing `cargo bloat` does is runs `cargo build`, so if you do not see anything means that `cargo build` frozen.

**Comment by Rudxain** [comment]
The crate only has 1 dependency, but that dep has **many** deps, and I hade this in my `Cargo.toml`:
```
[profile.release]
strip = true
lto = true
codegen-units = 1
```
I commented-out everything, and the latency reduced. Then I enabled everything (except `strip`, because of the symbol error) and the latency was the same. It seems `cargo` has generated a full cache for everything. I'll delete `target` to compare timings

**Comment by Rudxain** [comment]
Deleting `target` makes a difference, but it's unoticeable now... this is weird. I'll close the issue because everything seems fine. Maybe my HDD was in use while I ran the cmd? I didn't have any background processes using the disk. Maybe it was a system update? But the weird thing is that it happened **thrice** (2 in a row some hours ago, and once now)

**Comment by RazrFalcon** [comment]
Yes, your profile is very computation expensive. So this is expected. `lto` alone can take forever depending on the number of dependencies. 

Also note that `strip = true` isn't compatible with `cargo bloat`.

**Comment by acheronfail** [comment]
FWIW I also encountered this issue - it would block and appear frozen, but in my case even after waiting 4+ minutes nothing happened... 

Came across this issue and I also have been compiling with 

```rust
[profile.release]
strip = true
lto = true
codegen-units = 1
```

So I commented that out, ran `rm -rf target` and tried again. This next time, it hung for ~15 seconds and then began building.

**Comment by RazrFalcon** [comment]
The handing issue is on your side. `cargo bloat` doesn't do anything computation intensive.

**Comment by acheronfail** [comment]
Oh actually, I tried reproducing this since even with the aforementioned settings in Cargo.toml it wasn’t hanging. 

I think I figured it out: while it appears to “hang” or “block”, cargo is actually performing a download - it’s just that there’s no output to reflect this. 

Perhaps cargo is outputting the download progress differently? Or maybe it doesn’t output it when used in this way?

Either way. When it’s blocking here it seems to be downloading or updating the register, etc, and for some reason that progress isn’t surfaced.

**Comment by RazrFalcon** [comment]
All cargo output should be redirected. Are you sure you're using the latest version of `cargo bloat`?

**Comment by acheronfail** [comment]
Yep! Installed it with `cargo install cargo-bloat` just hours before I found this issue.

Since `cargo bloat` redirects all output, perhaps this is a issue with `cargo` itself. :thinking:

**Comment by RazrFalcon** [comment]
Maybe. It works just fine to me. Try running `cargo build` without `cargo bloat` and see if it works the same. Maybe it's your OS/terminal/something.

Here is the code: https://github.com/RazrFalcon/cargo-bloat/blob/8aa6c72ca93636c5aeeda9d95d308ba2b2fc700a/src/main.rs#LL571C25-L571C32
We simply pipe the output. Unless you use the `--time` flag.

**Comment by acheronfail** [comment]
So weird, I tried it in different terminals, and also on different machines and the result is the same.

Here's what I'm doing to reproduce it:

```bash
# install cargo bloat (use --force to re-install)
$ cargo install cargo-bloat

# clone a repo with a decent amount of dependencies
$ git clone https://github.com/BurntSushi/ripgrep.git
$ cd ripgrep

# try cargo bloat
# TIP: open a network monitor (like `nethogs` or whatever is good for your OS) and you'll see the download happening with no output
$ cargo bloat

# if I SIGINT `cargo bloat` with ^C, and then run
$ cargo build
# ... there is immediately output
```

**Comment by RazrFalcon** [comment]
So downloads are not visible, but _Compiling_ is?

**Comment by acheronfail** [comment]
> So downloads are not visible, but Compiling is?

That's right. I'm assuming this is only happening to myself and @Rudxain... You can see the downloads? :thinking:

**Comment by RazrFalcon** [comment]
I guess cargo writes it to `stderr`, which we ignore. Will test it.

**Comment by videni** [comment]
I also have this issue. No output, hanging forever. Actually, I once tried `cargo bloat` several times. It never works, and neither does it today.

```
# finished instantly without any error
cargo build 
# Hangle forever
cargo bloat --release -n 10  
```

I tried this crate in different Rust projects, since it never works, I thought it was a toy.
